src/Tools/BuildValidator/BuildConstructor.cs

```diff
@@ -232,57 +234,60 @@ private static (OptimizationLevel, bool) GetOptimizationLevel(string? optimizati
                 references: metadataReferences,
                 options: compilationOptions);
         }
 
-        private static VisualBasicCompilationOptions CreateVisualBasicCompilationOptions(CompilationOptionsReader optionsReader)
+        private static VisualBasicCompilationOptions CreateVisualBasicCompilationOptions(CompilationOptionsReader optionsReader, string fileName)
         {
             var pdbCompilationOptions = optionsReader.GetMetadataCompilationOptions();
 
-            var langVersionString = pdbCompilationOptions.GetUniqueOption("language-version");
-            pdbCompilationOptions.TryGetUniqueOption("optimization", out var optimization);
-            pdbCompilationOptions.TryGetUniqueOption("define", out var define);
-            pdbCompilationOptions.TryGetUniqueOption("strict", out var strict);
-            pdbCompilationOptions.TryGetUniqueOption("checked", out var checkedString);
+            var langVersionString = pdbCompilationOptions.GetUniqueOption(CompilationOptionNames.LanguageVersion);
+            pdbCompilationOptions.TryGetUniqueOption(CompilationOptionNames.Optimization, out var optimization);
+            pdbCompilationOptions.TryGetUniqueOption(CompilationOptionNames.Define, out var define);
+            pdbCompilationOptions.TryGetUniqueOption(CompilationOptionNames.GlobalNamespaces, out var globalNamespacesString);
+
+            IEnumerable<GlobalImport>? globalImports = null;
+            if (!string.IsNullOrEmpty(globalNamespacesString))
+            {
+                globalImports = GlobalImport.Parse(globalNamespacesString.Split(';'));
+            }
 
             VB.LanguageVersion langVersion = default;
             VB.LanguageVersionFacts.TryParse(langVersionString, ref langVersion);
 
             var preprocessorSymbols = string.IsNullOrEmpty(define)
                 ? Array.Empty<KeyValuePair<string, object>>()
-                : define.Split(';')
+                : define.Split(',')
                     .Select(s => s.Split('='))
                     .Select(a => new KeyValuePair<string, object>(a[0], a[1]))
                     .ToArray();
 
             var parseOptions = VisualBasicParseOptions.Default.WithLanguageVersion(langVersion)
                 .WithPreprocessorSymbols(preprocessorSymbols);
 
-            var (optimizationLevel, _) = GetOptimizationLevel(optimization);
-
-            bool.TryParse(checkedString, out var isChecked);
-            bool.TryParse(strict, out var isStrict);
+            var (optimizationLevel, plus) = GetOptimizationLevel(optimization);
+            var isChecked = OptionToBool(CompilationOptionNames.Checked) ?? true;
+            var embedVBRuntime = OptionToBool(CompilationOptionNames.EmbedRuntime) ?? false;
+            var rootNamespace = OptionToString(CompilationOptionNames.RootNamespaces) ?? null;
```

Comment:
`x ?? null` seems odd. Isn't that like `x != null ? x : null`?

Reply:
Yeah ... that is a bit silly